Added list reference (#2874)
conan-io/docs: reference/commands/list.rst

    Lists existing recipes, revisions or packages in the cache or in remotes given a complete
    reference or a pattern.

    positional arguments:
      reference             Recipe reference or package reference. Both can contain * as wildcard at any reference field. If
                            revision is not specified, it is assumed latest one.

    optional arguments:
      -h, --help            show this help message and exit
      -f FORMAT, --format FORMAT
                            Select the output format: json, html
      -v [V]                Level of detail of the output. Valid options from less verbose to more verbose: -vquiet, -verror,
                            -vwarning, -vnotice, -vstatus, -v or -vverbose, -vv or -vdebug, -vvv or -vtrace
      --logger              Show the output with log format, with time, type and message.
      -p PACKAGE_QUERY, --package-query PACKAGE_QUERY
                            Only list packages matching a specific query. e.g: os=Windows AND (arch=x86 OR compiler=gcc)
      -r REMOTE, --remote REMOTE
                            Remote names. Accepts wildcards
      -c, --cache           Search in the local cache

> The ``conan list`` command is a in-depth-oriented command, and it can lists recipes and packages

Author: ```suggestion
The ``conan list`` command is an in-depth-oriented command, and it can list recipes and packages
```

from the local cache or from any specified remotes. Depending on the patterns specified as argument,
it is possible to list:

* Recipe references.
* Recipe revisions.
* Package IDs and their configurations.
* Package revisions.

Use cases
---------

Show recipe references
**********************

.. code-block:: bash

    $ conan list *
    Local Cache:
      hello
        hello/1.0
        hello/2.0
      string-view-lite
        string-view-lite/1.6.0
      zlib
        zlib/1.2.11

.. code-block:: bash

    $ conan list zlib/*
    Local Cache:
      zlib
        zlib/1.2.11

Show recipe revisions
*********************

.. code-block:: bash

>     $ conan list zlib/1.2.11#*

Reviewer: I think maybe we should explain a bit here or at least link to other part of the documentation about the different parts that compose a Conan reference: `name/version@use/channel#rrev:pkgid#prev`

Author: Good point!

    Lists existing recipes, revisions or packages in the cache or in remotes given a complete
    reference or a pattern.

    positional arguments:
      reference             Recipe reference or package reference. Both can contain * as wildcard at any reference field. If
                            revision is not specified, it is assumed latest one.

    optional arguments:
      -h, --help            show this help message and exit
      -f FORMAT, --format FORMAT
                            Select the output format: json, html
      -v [V]                Level of detail of the output. Valid options from less verbose to more verbose: -vquiet, -verror,
                            -vwarning, -vnotice, -vstatus, -v or -vverbose, -vv or -vdebug, -vvv or -vtrace
      --logger              Show the output with log format, with time, type and message.
      -p PACKAGE_QUERY, --package-query PACKAGE_QUERY
                            Only list packages matching a specific query. e.g: os=Windows AND (arch=x86 OR compiler=gcc)
      -r REMOTE, --remote REMOTE
                            Remote names. Accepts wildcards
      -c, --cache           Search in the local cache

> The ``conan list`` command is an in-depth-oriented command, and it can list recipes and packages

Reviewer 2: "in-depth-oriented" in google gave a exclusive depth psychology books and mental health books 🙈 

```suggestion
The ``conan list`` command is an broad search command, and it can list recipes and packages
```

This one is me thinkings WAN... maybe someone has better non-networking word suggestions

```suggestion
The ``conan list`` command is an wide-area command, and it can list recipes and packages
```

from the local cache or from any specified remotes. Depending on the patterns specified as argument and taking into

> account that a complete Conan reference looks like this ``name/version@use/channel#rrev:pkgid#prev``,

Author: should it be enough @czoido ?

Reviewer: Maybe a bit more explanation with some examples of the patterns you can use and what is the expected output?

it is possible to list:

> * Recipe references (``name/version@use/channel```.

Reviewer: ```suggestion
* Recipe references (``name/version@user/channel``).
```

    reference or a pattern.

    positional arguments:
      reference             Recipe reference or package reference. Both can contain * as wildcard at any reference field. If
                            revision is not specified, it is assumed latest one.

    optional arguments:
      -h, --help            show this help message and exit
      -f FORMAT, --format FORMAT
                            Select the output format: json, html
      -v [V]                Level of detail of the output. Valid options from less verbose to more verbose: -vquiet, -verror,
                            -vwarning, -vnotice, -vstatus, -v or -vverbose, -vv or -vdebug, -vvv or -vtrace
      --logger              Show the output with log format, with time, type and message.
      -p PACKAGE_QUERY, --package-query PACKAGE_QUERY
                            Only list packages matching a specific query. e.g: os=Windows AND (arch=x86 OR compiler=gcc)
      -r REMOTE, --remote REMOTE
                            Remote names. Accepts wildcards
      -c, --cache           Search in the local cache

The ``conan list`` command is an wide-area command, and it can list recipes and packages

> from the local cache or from any specified remotes. Depending on the patterns specified as argument and taking into
> account that a complete Conan reference looks like this ``name/version@use/channel#rrev:pkgid#prev``,

Reviewer 3: ```suggestion
from the local cache or from any specified remotes. Depending on the type of pattern specified as argument and taking into
account that a complete Conan reference looks like this ``name/version@use/channel#rrev:pkgid#prev``,
```

> The ``conan list`` command is a wide-area command, and it can list recipes and packages
> from the local cache or from any specified remotes. Depending on the type of pattern specified as argument and taking into
> account that a complete Conan reference looks like this ``name/version@use/channel#rrev:pkgid#prev``,

Reviewer: ```suggestion
The ``conan list`` command can list recipes and packages
from the local cache or from any specified remotes. Depending on the type of pattern specified as argument and taking into
account that a complete Conan reference looks like this ``name/version@use/channel#rrev:pkgid#prev``,
```
I think we can omit that...
